### PR TITLE
Description of project lifecyle change requests and templates

### PR DESCRIPTION
**Pull request by gkunz**
This PR adds a description of the lifecycle change request process and corresponding templates for documenting the lifecycle change request. This is joined work with @lehors

Notes:
* The intention of the described process is to document lifecycle change requests in the TAC repository
* The process requires updating the project listing to ensure consistency. This listing is currently located in the README, but with a growing number of projects, this list should be moved to a separate file.
* The IP policy and license review process should be handled via a related GitHub issue. All results of the review should be made publicly available via this issue if possible. Question: is there any information which should not be made available, e.g., documents signing over trademark right for donated projects, etc. Input from LF is appreciated.

### CODE REVIEW

**Review by inferno-chromium — APPROVED** [review]

**Comment by lehors** [comment]
For what it's worth, we also explored using different Pull Request templates but unfortunately this is currently very poorly supported by GitHub. Unlike Issue templates for which GitHub provides a nice UI with a "template chooser", using different templates for PRs require the user to do some manual editing of the URL! This is clearly not a user friendly solution so we abandon this approach in favor of the one being proposed in this PR.

**Review by bobcallaway — COMMENTED** [review]
a couple nits, otherwise LGTM

File: process/project-lifecycle.md
Comment (annabellegoth2boss) [inline process/project-lifecycle.md:167]: "(or another interested party if no maintainer is available)"
Was this to accommodate Archived? If so, should we specify so it's not misinterpreted as non-maintainers could push a project from Incubation to Graduation without the maintainers' involvement?

File: process/project-lifecycle.md
Comment (annabellegoth2boss) [inline process/project-lifecycle.md:178]:
```suggestion
Contributing an existing project to the OpenSSF requires a one-time IP policy and license review with The Linux Foundation. To enable tracking of the IP policy and license review process, an issue in the TAC repo should be created. The results of the review as well as all relevant data (e.g., results of license scans) must be attached to this issue. Finally, the issue must be mentioned in the project creation PR and the latter can only be merged once the issue tracking the IP policy and license review has been closed.
```

**Review by annabellegoth2boss — COMMENTED** [review]
Overall looks good -- question in comments about if the more reviewer friendly workflow for project submissions is a PR or Issue

File: process/project-lifecycle.md
Comment (lehors) [reply process/project-lifecycle.md:167]: Yes, it is. Typically when a project is being archived there is no longer any maintainer around to do any such work so I thought we should provide for the possibility of having someone else. One possibility is to make this a community architect/LF staff member if you're concerned that this might be misused (although I can't imagine this being practically possible if the project is actively maintained).

File: process/project-lifecycle.md
Comment (lehors) [reply process/project-lifecycle.md:169]: Is "the issue format" really different? Why can't the discussion happen in comment to the PR, just like on the issue?
The proposed workflow aims at minimizing the number of steps involved. Once the PR is approved it is merged and we're done. Otherwise we will need to get both the issue closed and then the PR submitted, approved, and merged as a second step.

File: process/project-lifecycle.md
Comment (gkunz) [reply process/project-lifecycle.md:167]: Yes, @annabellegoth2boss, that was the intention and I can change the text to be more explicit. Another alternative used in LF projec is that a member of the TAC opens the PR to archive a project if no maintainer can be found to do this on behave of  the project.

Comment (annabellegoth2boss) [reply process/project-lifecycle.md:167]:
```suggestion
For initiating the creation of a new project or for requesting a change of a project's lifecycle stage, an application must be submitted to the TAC. To this end, a maintainer of the project (or any OpenSSF member in the case of Archived projects) creates a PR in this repository with the following changes:
```

File: process/project-lifecycle.md
Comment (annabellegoth2boss) [reply process/project-lifecycle.md:169]: Discussion can certainly happen in a PR. 

I'm thinking of #101 where there's been robust discussion, and having that linear Issue format is helpful for readability. I'd prefer to design for accessible and digestible communication over time-to-approval on this one. 

cc @AevaOnline who might have thoughts from the CCC process as well.

Comment (lehors) [reply process/project-lifecycle.md:169]: But if you comment at the bottom of a PR it is similarly linear, isn't it?

Comment (TheFoxAtWork) [reply process/project-lifecycle.md:169]: So it depends on the lifecycle stage (within the CNCF).

For brand new projects applying to sandbox, its a form they fill out and we review.  This could be reasonably managed as an Issue template. But not all projects need to go through Sandbox, some can apply directly to incubation.

For Applying to Incubation, or changing from one stage to another (sandbox to incubation, incubation to graduation) its a PR. We keep [a template in the repo](https://github.com/cncf/toc/blob/main/process/graduation-proposal-template.md) that projects copy into their PR.  Discussion can occur on the PR in the particular content areas, but it still provides consistency in content. When accepted, they are merged in. Such as with [Linkerd](https://github.com/cncf/toc/pull/616), or [knative](https://github.com/cncf/toc/pull/762/files).

My recommendation: 
To be consistent in the processes and experience of projects, I would establish a template for each pursued stage and ask projects to copy the content into their PR to serve as the template. So one template for Sandbox, one for incubating, and one for graduation. The template should be clear in the information expected and its content sections should solicit meaningful information from the project that can drive an informed decision on whether they can move to the stage requested, or provide enough information to show why they cannot yet move. The deciding body should be able to clearly point to information in the PR, as well as other supporting materials, to justify their decision and provide recommendations actions back to the project based on that decision with a path forward.

Discussions then occur in the PR. The only outstanding process-oriented gotcha then would be the assignment of a sponsor to it. This should be mitigated by having the assigned or volunteered sponsor suggest a change to the PR for their handle in the otherwise empty "sponsor" section, and having the maintainer commit the suggestion.

File: process/templates/PROJECT_NAME_sandbox_stage.md
Comment (TheFoxAtWork) [inline process/templates/PROJECT_NAME_sandbox_stage.md:13]:
```suggestion
  * "yes / no / not applicable. If yes, provide a link to the corresponding GitHub issue."
  
  ### Project References
  The project should provide a list of existing resources with links to the repo, website, a roadmap, contributing guide, demos and walkthroughs, and other material to showcase the existing breadth, maturity, and direction of the project.
 | Reference | URL |
|---|---|
| Repo |   |
| Website |   | 
| Contributing |   |
| Roadmap |   | 
| Demos |    |
| Other |    |

**Review by TheFoxAtWork — COMMENTED** [review]
Provided two comments, @annabellegoth2boss i believe i addressed your question, let me know if you need more detail.  I also recommended an addition to the sandbox stage as this will likely be the first introduction of a project to the OpenSSF members and can provide a consolidated place to assist in the review for acceptance, but i can also see it included in the others for brevity.

File: process/templates/PROJECT_NAME_sandbox_stage.md
Comment (gkunz) [reply process/templates/PROJECT_NAME_sandbox_stage.md:13]: Given the discussion in #105, i.e., project can directly join at the incubation stage, I suggest to add the requirement for an IP policy review and the table of references to the incubation template file as well. Objections?

File: process/project-lifecycle.md
Comment (gkunz) [reply process/project-lifecycle.md:169]: Thanks for the feedback, @TheFoxAtWork

**Review by TheFoxAtWork — COMMENTED** [review]
LGTM!!! Thanks so much @gkunz !

**Comment by inferno-chromium** [comment]
Looks like comments are addressed, merging.

**Pull request merged**
